int32 times
---
bool ack
string msg
---
int32 times_completed